dotCMS/src/main/java/com/dotcms/ai/client/AIResponseEvaluator.java

```diff
@@ -0,0 +1,36 @@
+package com.dotcms.ai.client;
+
+import com.dotcms.ai.domain.AIResponseData;
+
+/**
+ * Interface for evaluating AI responses.
+ * It provides methods to process responses and exceptions, updating the provided metadata.
+ *
+ * <p>Methods:</p>
+ * <ul>
+ *   <li>\fromResponse\ - Processes a response string and updates the metadata.</li>
+ *   <li>\fromThrowable\ - Processes an exception and updates the metadata.</li>
+ * </ul>
+ *
+ * @author vico
+ */
+public interface AIResponseEvaluator {
```

Comment:
this seems to be very couple with the AIClient, may be should be an inner interface into the AIClient so the client may be aware to the necessity to implement both interfaces